### PR TITLE
feat:modal

### PR DESCRIPTION
**Pull request by svetikbaihe**

### CODE REVIEW

File: src/modules/Cart/CartBody/component.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                }
+                onClick={handleDeleteProduct(product.id)}
+                className={styles.productDeleteBtn}
+              />
+            </div>
+            <hr />
+          </li>
+        )
+      })}
+    </ul>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx:45]: `      <ul>
        {cart.map(product => (
          <li key={product.id} className={styles.liProduct}>
            {isMobile && (
              
            )}
            <div className={styles.nameAmountPriceContainer}>
              <Text className={styles.productName}>
                {isMobile
                  ? truncateString(product.name, 10)
                  : truncateString(product.name, 20)}
              </Text>
              <div>
                <Text className={styles.productQuantity}>
                  {product.amount + 'x'}
                </Text>
                <Text className={styles.productPrice}>
                  {'@ ' + '$' + product.price.toFixed(2)}
                </Text>
              </div>
            </div>
            <Text className={styles.individualTotalPrice}>
              {'$' + (product.amount * product.price).toFixed(2)}
            </Text>
            <hr className={styles.hr} />
          </li>
        ))}
      </ul>
`
It looks better =)

File: src/modules/Cart/CartBody/component.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  return (
+    <ul>
+      {cart.map(product => {
+        return (
+          <li key={product.id} className={styles.liProductPosition}>
+            <div className={styles.cartInfoBlock}>
+              <div>
+                <Text className={styles.productName}>{product.name}</Text>
+                <div className={styles.priceQuantity}>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx:20]: Add space between lines on the same level

File: src/modules/Cart/CartBody/component.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                  <Text className={styles.productQuantity}>
+                    {product.amount + 'x'}
+                  </Text>
+                  <Text className={styles.productPrice}>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartBody/component.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                    {'@ ' + '$' + product.price.toFixed(2)}
+                  </Text>
+                  <Text className={styles.individualTotalPrice}>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartBody/component.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                    {'$' + (product.amount * product.price).toFixed(2)}
+                  </Text>
+                </div>
+              </div>
+              <Button
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartBody/component.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                type="text"
+                icon={
+                  <CloseCircleOutlined className={styles.productDeleteIcon} />
+                }
+                onClick={handleDeleteProduct(product.id)}
+                className={styles.productDeleteBtn}
+              />
+            </div>
+            <hr />
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartFooter/component.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+const { Text } = Typography
+
+const CartFooter: React.FC = () => {
+  const { cartTotalPrice, handleOpenModal } = useContainer()
+  return (
+    <div>
+      <div className={styles.orderTotalSumContainer}>
+        <Text className={styles.orderTotalTitle}>Order Total</Text>
+        <Text className={styles.orderTotalSum}>{'$' + cartTotalPrice}</Text>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartFooter/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartFooter/component.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      </div>
+      <div className={styles.carbonNeutralContainer}>
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartFooter/component.tsx]: Add space between lines on the same level

File: src/modules/Cart/CartFooter/component.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+          src="/icons/icon-carbon-neutral.svg"
+          className={styles.carbonImg}
+        />
+        <p>
+          This is a{' '}
+          <span className={styles.carbonAccentText}>carbon-neutral</span>{' '}
+          delivery
+        </p>
+      </div>
+      <Button
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartFooter/component.tsx]: Add space between lines on the same level

File: src/modules/ConfirmedOrder/OrderBody/component.tsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+                </div>
+                <Text className={styles.individualTotalPrice}>
+                  {'$' + (product.amount * product.price).toFixed(2)}
+                </Text>
+              </li>
+              <hr className={styles.hr} />
+            </>
+          )
+        })}
+      </ul>
```
Comment (ArtemPapusha) [inline src/modules/ConfirmedOrder/OrderBody/component.tsx]:
```
 <ul>
        {cart.map(product => (
          <li key={product.id} className={styles.liProduct}>
            {isMobile && (
              <img
                src={product.image.thumbnail}
                className={styles.orderProductImg}
              />
            )}
            <div className={styles.nameAmountPriceContainer}>
              <Text className={styles.productName}>
                {isMobile
                  ? truncateString(product.name, 10)
                  : truncateString(product.name, 20)}
              </Text>
              <div>
                <Text className={styles.productQuantity}>
                  {product.amount + 'x'}
                </Text>
                <Text className={styles.productPrice}>
                  {'@ ' + '$' + product.price.toFixed(2)}
                </Text>
              </div>
            </div>
            <Text className={styles.individualTotalPrice}>
              {'$' + (product.amount * product.price).toFixed(2)}
            </Text>
            <hr className={styles.hr} />
          </li>
        ))}
 </ul>
 ```

File: src/modules/Cart/CartBody/component.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+const { Text } = Typography
+
+const CartBody: React.FC = () => {
+  const { cart, handleDeleteProduct } = useContainer()
+
+  return (
+    <ul>
+      {cart.map(product => {
+        return (
```
Comment (ArtemPapusha) [inline src/modules/Cart/CartBody/component.tsx]: See order body example of comment.

File: src/modules/Cart/hook.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+import { Context } from '@state/context/ContextProvider'
+
+const useContainer = () => {
+  const context = useContext(Context)
+
+  const cart = context.state.cart
+
+  const handleCloseModal = () => {
+    context.closeModal()
+  }
```
Comment (ArtemPapusha) [inline src/modules/Cart/hook.ts]: Use hook useCallback for transporting like prop.

File: src/modules/DessertMenu/DessertCard/component.tsx [hunk 13]
```diff
@@ -8,60 +8,82 @@ import {
...
+  const handlePlus = () => {
+    handleIncrement({ id: item.id })
   }
 
-  const decrement = () => {
-    setCount(prevCount => prevCount - 1)
+  const handleMinus = () => {
+    handleDecrement({ id: item.id })
+    handleDeleteZeroAmount()
   }
```
Comment (ArtemPapusha) [inline src/modules/DessertMenu/DessertCard/component.tsx]: Move it into useContainer.

File: src/modules/DessertMenu/DessertCard/component.tsx [hunk 14]
```diff
@@ -8,60 +8,82 @@ import {
...
-    setCount(prevCount => prevCount + 1)
+  const productAmount = foundProduct ? foundProduct.amount : 0
+
+  const handleAddToCart = () => {
+    handleAddProductToCart(item)
+  }
+
+  const handlePlus = () => {
+    handleIncrement({ id: item.id })
   }
```
Comment (ArtemPapusha) [inline src/modules/DessertMenu/DessertCard/component.tsx]: Move it into useContainer.

File: src/modules/Cart/CartBody/component.tsx (hunk 2 above)
Comment (svetikbaihe) [reply src/modules/Cart/CartBody/component.tsx:20]: done

File: src/modules/Cart/hook.ts (hunk 12 above)
Comment (svetikbaihe) [reply src/modules/Cart/hook.ts]: everything's been modified. thank you!

File: src/modules/Cart/hook.ts [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+  const handleOpenModal = useCallback(() => {
+    context.openModal({
+      type: 'ORDER_MODAL',
+      modalProps: null,
+    })
+  }, [])
+
+  const handleDeleteProduct = useCallback((id: number) => () => {
+    context.deleteProduct(id)
+  }, [cart])
```
Comment (ArtemPapusha) [inline src/modules/Cart/hook.ts]: Clear dependencies.

File: src/modules/DessertMenu/DessertCard/hook.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+  const context = useContext(Context)
+
+  const cart = context.state.cart
+
+  const foundProduct = cart?.find(product => product.id === item?.id)
+  const productAmount = foundProduct ? foundProduct.amount : 0
+
+  const handleIncrement = useCallback(() => {
+    context.incrementProduct({ id: item?.id })
+  }, [cart])
```
Comment (ArtemPapusha) [inline src/modules/DessertMenu/DessertCard/hook.ts]: Change on item instead cart.

File: src/modules/DessertMenu/DessertCard/hook.ts [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+
+  const handleDecrement = useCallback(() => {
+    context.decrementProduct({ id: item?.id })
+    context.deleteZeroAmount()
+  }, [cart])
```
Comment (ArtemPapusha) [inline src/modules/DessertMenu/DessertCard/hook.ts]: Change on item instead cart.

File: src/modules/DessertMenu/DessertCard/hook.ts [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+
+  const handleAddProductToCart = useCallback(() => {
+    context.addProductToCart(item)
+  }, [cart])
```
Comment (ArtemPapusha) [inline src/modules/DessertMenu/DessertCard/hook.ts]: Change on item instead cart.

**Review by ArtemPapusha — APPROVED** [review]

**Pull request merged**